### DISCUSSION TITLE
Whitelists not affecting subdomains

### DISCUSSION THREAD

**Discussion by agneevX** [AdguardTeam/AdGuardHome#3222] [Q&A] [answered]
### Prerequisites

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details

* Version of AdGuard Home server: 0.106.3
* How did you install AdGuard Home: GH Releases
* How did you setup DNS configuration: N/A
* If it's a router or IoT, please write device model: N/A
* CPU architecture: N/A
* Operating system and version: N/A

### Expected Behavior

When a whitelist has a following format, it should automatically apply to all subdomains.

```
akamaihd.net
```

Meaning `cdn.akamaihd.net` should also be allowed, except it does not.

### Actual Behavior

Subdomains are getting blocked, since they exist in a blocklist.

<img width="435" alt="image" src="https://user-images.githubusercontent.com/19761269/120453310-8ea88000-c3b0-11eb-89b3-54b4cf38242b.png">

Is there a priority between blocklists, whitelists and custom rules that could be affecting this?

**Comment by TheCeet** [comment, marked as answer]
When you read the filter rules info, you'll need to make it like:
`@@||akamaihd.net`
See: https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#examples

**Reply by agneevX** [reply]
I'm a bit confused. This is part of an added whitelist, not a custom rule: https://github.com/agneevX/whitelist/blob/master/whitelist.txt#L11

**Reply by ameshkov** [reply]
You have two options:

1. Use this notation `||akamaihd.net^` and add the list to "DNS allowlists"
2. Use `@@||akamaihd.net^`, this way it does not matter where you add it (allowlists or blocklists), AdGuard Home will understand it properly anyway.

**Reply by agneevX** [reply]
So without the notations, it's not supported? 
Meaning, will I have to change the format for every domain in the whitelist?

**Reply by ameshkov** [reply]
If you want subdomains covered, then yes, you should use the rules notation, not just a list of domains.

Consider using https://github.com/AdguardTeam/HostlistCompiler to convert rules.
